### ISSUE TITLE
The Issues 0.14 Causes to DragonProxy (Post issues here)

### ISSUE THREAD

**Issue by RandomAltThingy** [DragonetMC/DragonProxy#91]
Alright people of @DragonetMC/developers, listen up.

The 0.14.0 beta has been released. We will need to find out what features it has and implement them here. If you have bugs running with 0.14 then tell here. The 0.14 development starts **now**.

**Comment by DefinitlyEvil** [comment]
Any protocol updates?

**Comment by DefinitlyEvil** [comment]
@RandomAltThingy If you have Android APK please send me so I can decompile it.

**Comment by RandomAltThingy** [comment]
I have an apk

**Comment by RandomAltThingy** [comment]
@DefinitlyEvil Just go to YouTube search *"mcpe 0.14.0 apk"* and click on "Today" so it will search today's uploads.

**Comment by RandomAltThingy** [comment]
@DefinitlyEvil https://db.tt/SQK99V7S

**Comment by Pet0203** [comment]
Dispensers and red sandstone blocks/slabs/stairs has been added. If I'm right, we removed the dispensers manually? Also all the other red stone materials has been added. "New item rendering system", can this be something important? More about what has been added & changed can be found here http://minecraft.gamepedia.com/Pocket_Edition_Alpha_0.14.0

**Comment by RandomAltThingy** [comment]
@Pet0203 Rendering is done on the client. Also, I know what mcpe contains I have it.

**Comment by RandomAltThingy** [comment]
PC item frames ID is 389. For PE there is ids 388 and 390, but no 389. Item frames in PE are block entities and I cannot find their ID on the wiki.

**Comment by RandomAltThingy** [comment]
@DefinitlyEvil Create a repo called **DragonProxy-Android**, I'm going to start porting it

**Comment by DefinitlyEvil** [comment]
@RandomAltThingy Sure, and it will be kinda easy to port it since there is no native lib required for either DragonProxy or its dependencies. xD And make sure you use sub-modules instead of copying all files. xD

**Comment by RandomAltThingy** [comment]
How would you use submodules in an android app?

And what should we do about the dependencies in the pom.xml? Android doesn't support maven unless it's a local repo.

**Comment by DefinitlyEvil** [comment]
@RandomAltThingy Hmm... We can add a sub-module and create a separate folder for Android app, then include the source folder to the app and manually import dependencies. Also from the Android app, we should override something like console output.

**Comment by TekkitCommando** [comment]
Why are you making an Android app?

**Comment by DefinitlyEvil** [comment]
@TekkitCommando People are expecting to use DragonProxy without turning their computer on. xD 
@RandomAltThingy Make sure u used Service thing to run the server or it will be unstable. ;)

**Comment by DefinitlyEvil** [comment]
@TekkitCommando People are expecting to use DragonProxy without turning their computer on. xD 
@RandomAltThingy Make sure u used Service thing to run the server or it will be unstable. ;)

**Comment by TekkitCommando** [comment]
@DefinitlyEvil could you please explain a little more I'm still not see how this is a high priority when DragonProxy has a lot of bugs

**Comment by DefinitlyEvil** [comment]
@TekkitCommando Porting to Android will not take much time becuase Java is cross-platform.

**Comment by RandomAltThingy** [comment]
I haven't done anything yet as ive been asleep (its 2am xD) and I still need to eat and I think I have to go out today

**Comment by TekkitCommando** [comment]
Yea but the API and UI will make it take longer, but I trust you know what you are doing

**Comment by DefinitlyEvil** [comment]
@RandomAltThingy @TekkitCommando Well take your time making it guys, and let's talk on Gitter or IRC. xD

**Comment by RandomAltThingy** [comment]
IRC FTW

**Comment by seria2** [comment]
PC player don't show in MCPE (0.14.3, latest 250 build)

**Comment by RandomAltThingy** [comment]
@seria2 If you can, download the latest [pre-release](https://github.com/dragonetmc/dragonproxy/releases) then join using 0.13.1 and see if the bug exists.

**Comment by RandomAltThingy** [comment]
@DefinitlyEvil I fixed the build with placeholders and useless functions.

**Comment by DefinitlyEvil** [comment]
@RandomAltThingy Okay. ;)

**Comment by Lohkdesgds** [comment]
When will be available a 0.13.2 version to connect to 1.8.*? And please, don't stop the PC version ;-;
I use it to play with my friends (me on PC hosting everything)

**Comment by DefinitlyEvil** [comment]
@Lohkdesgds READ THE CLOSED ISSUES PLEASE. And we don't need to support PC/Mac to connect because you can use Spigot+DragonProxy combination and which is much better than Glowstone-based Dragonet.

**Comment by RandomAltThingy** [comment]
@DefinitlyEvil Can I have the website code for creating an account, logging in, logging out etc please? I am working on a clan website for MCPE but I have no idea how to do login.

**Comment by DefinitlyEvil** [comment]
@RandomAltThingy You don't want too see the code, it's a mess.

**Comment by RandomAltThingy** [comment]
@DefinitlyEvil Messy code is what I'm good at

**Comment by RandomAltThingy** [comment]
Image of the homepage on mobile: http://m.imgur.com/AfOk1FM

**Comment by DefinitlyEvil** [comment]
@RandomAltThingy Go to IRC.

**Comment by RandomAltThingy** [comment]
K

**Comment by kniffo80** [comment]
I'm a JAVA developer for more than 10 years now. I just forked the Repo and tried to build it. Anyway, having problems with those Lombok Annotations and - the code does not work out of the box, there are compiler errors.

Did anyone forget to commit something?

Why did i try to compile it for myself? Because the client I'm using is 0.13.2. At the moment with the latest release, the client cannot connect - because of the version.

Maybe, i can contribute - if i have time.

**Comment by RandomAltThingy** [comment]
I think it is not working because of the new config code, but we are busy so cant fix. Feel free to fix it.

**Comment by kniffo80** [comment]
Thanks @RandomAltThingy - will try to fix it. Anyway, i'll remove those Lombok annotation stuff. Because if removed, it's compilable from scratch.

And yes, it's because of the new config code ;)

**Comment by DefinitlyEvil** [comment]
@kniffo80 Use the code from latest release instead of the master branch or it may fail.

**Comment by kniffo80** [comment]
@DefinitlyEvil thanks for the answer!

**Comment by DefinitlyEvil** [comment]
@kniffo80 Do not use this place as a forum. if you have questions, go to http://forums.dragonet.org to ask.

**Comment by RandomAltThingy** [comment]
The issues is for asking questions... As long as it doesn't turn from a QnA into a conversation

**Comment by seanth** [comment]
Where does one get a 0.14 beta build?

**Comment by Pet0203** [comment]
0.14.0 has been released for IOS

**Comment by TheDiamondYT1** [comment]
@DefinitlyEvil I still need the website code.

**Comment by DefinitlyEvil** [comment]
@TheDiamondYT1 yeah yeah, wait for me I'm gonna grab it.

**Comment by TheDiamondYT1** [comment]
GITTER

**Comment by robske110** [comment]
Block break lag
not your fault GENERAL( pocketmine)

**Comment by TheDiamondYT1** [comment]
MCPE Bug.

**Comment by DefinitlyEvil** [comment]
@robske110 Nice to hear that. :D #BlameMojang

**Comment by robske110** [comment]
Did you hear of the Win10 client. We on pocketmine/ClearSky are just crying because we don't understand what that client sends for a shit (Especially Inventory)

**Comment by TheDiamondYT1** [comment]
How about 'DragonetMC - A BytePowered Company' :)

**Comment by DefinitlyEvil** [comment]
@robske110 We'll figure it out. 
@TheDiamondYT1 DragonetMC is just a trademark not a company.

**Comment by robske110** [comment]
The real problem is: We can't check if its win10 or ios/android. So we could fox win10 but then ios+android won't work anymore. Please let me know if you made a breakthrough here

**Comment by TheDiamondYT1** [comment]
priority is ios/android.

**Comment by robske110** [comment]
Yes, and this is why we rage out at PM, we knoe we shouldn't break ios/android but want Win10 to work

**Comment by iMelsom** [comment]
Hi. 
Love Dragonet proxy. Unfortunately I got it up and running qith the 1.14.0 beta almost exactly a couple of days before Mojang updated the clients to 1.14.1.

I see that the 1.14.1 client happily connects to a 1.14.0 Pocketmine (Imagicalmine) server, so any chance of quick-fixing Dragonet proxy to accept 1.14.1 clients as well? 

(1.14.1 is only a skin update: http://minecraft.gamepedia.com/Pocket_Edition_Alpha_0.14.1)

**Comment by robske110** [comment]
And a bug fix update
And it's protocol 47 now

**Comment by DefinitlyEvil** [comment]
@iMelsom Dragonet isn't ready for you to connect to PE servers yet. xD

**Comment by iMelsom** [comment]
@DefinitlyEvil : Yes, I know that, and I may have been a bit vague in my question. 

I meant that since the clients are able to connect to a 1.14.0 pocektmine server as a 1.14.1 client, the changes Mojang did cannot be too big, and therefore it ought to follow that it should be possible for a client to connect to a 1.14.0 Dragonproxy connected to a 1.8.9 vanilla server. It may be as simple as adjusting the version number allowed to allow 1.14.1 clients to connect ?

What happends is that the clients are thrown out with the error message outdated client.

(that by the way are another bug: Dragonproxy do not differ between the two different errors "outdated client" and "Outdated server", and only throws a "outdate client", also when a 1.14.1 client connects to the curretn 1.14.0 dragon proxy)

**Comment by DefinitlyEvil** [comment]
@iMelsom Okay I got it.

**Comment by TheDiamondYT1** [comment]
@DefinitlyEvil I need help making a decision on whether to make a C++ server software or Java.

**Comment by TheDiamondYT1** [comment]
Fuck it I'm doing java.

**Comment by TheDiamondYT1** [comment]
Can you give me admin for a bit so I can sort out stuff on the Dragonet repo such as branches etc. I want to move Dragonet-dusty back to Dragonet I've had an idea on how I will do this.

**Comment by iMelsom** [comment]
Testing the 1.14.1 friendly version now. list of things that do not work quite right yet: 

- Death is not handled correct. MCPE client has to be forcefully killed and restarted to respawn. (on android)
- on several users online I occationally experience failure to load more blocks that the immediate surroundings. 
- I get a lot of java.lang.ClassCastExceptions

[logs.zip](https://github.com/DragonetMC/DragonProxy/files/233361/logs.zip)

**Comment by TheDiamondYT1** [comment]
@iMelsom Are you using later build?

**Comment by TheDiamondYT1** [comment]
@iMelsom Latest*

**Comment by iMelsom** [comment]
Yep. Downloaded #298 a few minutes ago from circleCI

**Comment by TheDiamondYT1** [comment]
Hmmm. I thought the config thing was fixed.

**Comment by iMelsom** [comment]
Oh, the config thing is fixed, and I do manage to log on now, it seems to be a error in casting between protocols: 
java.lang.ClassCastException: java.lang.Byte cannot be cast to org.spacehq.mc.protocol.data.game.ItemStack

**Comment by DefinitlyEvil** [comment]
@iMelsom Very useful! thx

**Comment by Creeperface01** [comment]
and chunks dont load :/

servers:
play.survival-games.cz
play.gameteam.cz

**Comment by robske110** [comment]
This is because if the missing chunk radius packet

**Comment by iMelsom** [comment]
Found another bug:
-wind10 PE beta & android: 
In creative mode all pickable blocks are gone in the usable block tabs in the inventory. It is completely empty.

**Comment by Creeperface01** [comment]
I think chunk loading is the most important

**Comment by iMelsom** [comment]
@Creeperface01: I agree, but that doesn't mean that we shouldn't list any other bugs or faults we encounter.

**Comment by DefinitlyEvil** [comment]
Chunks over height 128 will not work, if spawn point is set higher than 128, clieny will shake like crazy.

**Comment by Pet0203** [comment]
Take a look at #17 
There are a few bugs there.

**Comment by DefinitlyEvil** [comment]
@Pet0203 Of course we can't fix height issue, so we should just focus on our own problems.

**Comment by Creeperface01** [comment]
So we must wait for 0.15

**Comment by TheDiamondYT1** [comment]
On 0.15 height will be 256 :)

**Comment by DefinitlyEvil** [comment]
nice

**Comment by LordOfQwerty** [comment]
I'm sure that if you can make PE data to PC data, you can make it so both ID's are used, but different depending on what version of Minecraft used. But I'm not sure if it can do that.

**Comment by QwertyPrism** [comment]
I am also sure, if my memory helps me, that Minecraft PE and PC armor values are different too

**Comment by QwertyPrism** [comment]
And about bug #5, the same problems can happen regularly on Minecraft PE Pocketmine-Based servers.

**Comment by QwertyPrism** [comment]
Not the one that the link goes to, but the one above

**Comment by QwertyPrism** [comment]
I'm sorry that I am commenting so much

**Comment by DefinitlyEvil** [comment]
It's okay dude, comments are appreciated. :D 
But I recommend u to squeeze those comments into a single one. xD

**Comment by VillagerCube** [comment]
Ok, I'm reading all my notifications and responding to them, and I will squeeze them

**Comment by VillagerCube** [comment]
I would like to test this for issues, but I need to find the latest version first.

**Comment by DefinitlyEvil** [comment]
@VillagerCube Use CircleCI, login as Github user then download.

**Comment by VillagerCube** [comment]
Ok thx.

**Comment by VillagerCube** [comment]
Testing for bugs in a little bit, busy at the moment, typing this on my iPhone, etc. But I'm back home, where I have a good computer and Minecraft PE.
